### ISSUE TITLE
Reverse match convention of translate exclusion patterns

### ISSUE THREAD

**Issue by facelessuser** [facelessuser/wcmatch#51]
One thing I regret was outputting exclusion patterns in translate as matches that are `True` if the folder is not excluded. This utilizes lookaheads to first check for anything that would exclude the file, and then grabs the filepath. This is slower than if we had just had a match (`True`) exclude the file.

Yes it is a clever pattern, but performance wise, it is not ideal. Internally, we are using an internal flag to force them in the better, faster orientation. I'm thinking we should just tear off the bandage in 4.0 and just output them in the better, faster way. Then we can get rid of the internal flag.

3.0 and 4.0 has really been a focus of getting things the way we ultimately want things. Fixing overlooked issues, and making course corrections before more people start using the library. I would have preferred to do this all in 3.0, but I didn't actually expect to have the motivation to move through 3.0 and 4.0 so fast, or even realize the full scope of what I wanted to do.

I imagine most people are using glob or fnmatch directly, and are not using the translate API. I also imagine the number of people using this library is quite small.

This will not affect Rummage or pyspelling, our file search and replace tool and document spellchecker, which is the biggest impact to me personally. Translation is really only used by me for internal stuff, and as I mentioned, we are forcing it the way we want it with a special internal flag.

In short, I'm willing to break API to work the better way, and possibly anger a very small few, if any, to get things to where we want them moving forward.  I'm hoping with the course corrections that took place in 3.0 and 4.0, that we won't have to make any drastic changes moving forward.
